classes/class-wc-connect-shipping-label.php

```diff
@@ -362,6 +364,14 @@ protected function get_states_map() {
 		}
 
 		public function should_show_meta_box() {
+			if ( null === $this->show_metabox ) {
```

Comment:
are you sure `calculate_should_show_meta_box()` is run twice? What should happen is that `should_show_metabox()` and `calculate_should_show_meta_box()` are both run one time, and then the second time only `should_show_metabox` is run.

Note: If might be that the pointer code gets executed before the render of the metabox, so it's possible that our usage here DOES result in `calculate_` getting run, but then when the calculation is skipped in the meta_box render code.